Guides/Trim.md

```diff
@@ -0,0 +1,116 @@
+# Trim
+
+[[Source](https://github.com/apple/swift-algorithms/blob/main/Sources/Algorithms/Trim.swift) |
+ [Tests](https://github.com/apple/swift-algorithms/blob/main/Tests/SwiftAlgorithmsTests/TrimTests.swift)]
+
+Returns a `SubSequence` formed by discarding all elements at the start and end of the Collection
+which satisfy the given predicate.
+
+```swift
+let results = [2, 10, 11, 15, 20, 21, 100].trimmed(where: { $0.isMultiple(of: 2) })
```

Comment:
Just a teeny bit of bikeshedding. Because trailing closure syntax causes `where:` to be omitted, there can be a little bit of ambiguity as to whether the closure determine's what's included or what's excluded. It so happens that, in English, "trimming" can take a direct object, while "trimmed" cannot, such that `[1, 2, 3].trimming { $0 == 1 }` suggests that every element equal to one is being dropped a little more strongly than "trimmed" would.

Reply:
Yeah, that sounds good. I like `trimmed`.